commands/charm/spinner/spinner.go

```diff
@@ -0,0 +1,95 @@
+package spinner
+
+import (
+	"fmt"
+	"os"
+
+	s "github.com/charmbracelet/bubbles/spinner"
+	tea "github.com/charmbracelet/bubbletea"
+)
+
+type SpinningLoader struct {
+	model   s.Model
+	prog    *tea.Program
+	cancel  bool
+	Message string
+}
+
+type Option func(*SpinningLoader)
+
+// New creates a new spinning loader.
+func New(opts ...Option) SpinningLoader {
+	sm := s.New()
+	sm.Spinner = s.Dot
+
+	l := SpinningLoader{
+		model:   sm,
+		Message: "",
+	}
+
+	for _, opt := range opts {
+		opt(&l)
+	}
+	return l
+}
```

Comment:
let's unexport the message field and make it a required argument on New:

```suggestion
type SpinningLoader struct {
	model   s.Model
	prog    *tea.Program
	cancel  bool
	message string
}

type Option func(*SpinningLoader)

// New creates a new spinning loader.
func New(message string, opts ...Option) SpinningLoader {
	sm := s.New()
	sm.Spinner = s.Dot

	l := SpinningLoader{
		model:   sm,
		message: message,
	}

	for _, opt := range opts {
		opt(&l)
	}
	return l
}
```